File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        productFromLeft = [1]*len(nums)
        productFromLeft[0] = nums[0]
        for i in range (1, len(nums)):
            productFromLeft[i] *= productFromLeft[i-1]*nums[i]

        productFromRight = [1]*len(nums)
        productFromRight[len(nums)-1] = nums[len(nums)-1]
        for i in range (len(nums)-2, -1, -1):
            productFromRight[i] *= productFromRight[i+1]*nums[i]
        ans = [1] * len(nums)

        for i in range (0, len(nums)):
            if(i==0): ans[i] *= productFromRight[i+1]
            elif(i==len(nums)-1): ans[i] *= productFromLeft[i-1]
            else: ans[i] *= productFromLeft[i-1]*productFromRight[i+1]
        return ans

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int ans = 0;
        int j = 0;
        for(int i=0;i<n;i++){
            if(i>0 && nums[i]==nums[i-1]) continue;
            nums[j] = nums[i];
            j++;
        }
        ans = j;
        return ans;
    }
};